core/src/main/java/org/jruby/RubyIO.java

```diff
@@ -1441,12 +1450,42 @@ public IRubyObject write(ThreadContext context, IRubyObject[] args) {
         return RubyFixnum.newFixnum(context.runtime, acc);
     }
 
-    final IRubyObject write(ThreadContext context, int ch) {
+    /**
+     * Write a single byte to this IO's write target.
+     *
+     * @param context the current context
+     * @param ch the byte to write, as an int
+     * @return the count of bytes written
+     */
+    public final IRubyObject write(ThreadContext context, int ch) {
```

Comment:
:+1: would like to rather see a `public void write(int b)` (without the context for Java consumers)

Reply:
This was actually just a protected method before, but I agree having non-context forms would be a good idea. In fact, I was thinking RubyIO could basically implement the same methods as Input/OutputStream if we decide on how to handle incoming bytes with no encoding (#6588).